include/platform/init.h

```diff
@@ -14,6 +14,39 @@
  * all of the required GPIO for the platform has been
  * initialized.
  */
-void  init_gpio(void);
+void init_valve_gpio(void);
+void init_gpio_clk();
+
+void init_alarm_led();
+void init_alarm_buzzer();
+void init_alarm(void);
+
+/**
+ * Initialize the I2C peripheral
+ * of the microprocessor. When this function completes
+ * all of the the I2C bus for the platform has been
+ * initialized.
+ */
+void init_i2c(void);
+/**
+ * Initialize an ADC (1) peripheral
+ * of the microprocessor. When this function completes
+ * an ADC for the platform has been initialized.
+ */
+void init_adc1(void);
+/**
+ * Initialize the uart peripheral
+ * of the microprocessor. When this function completes
+ * the uart (used for serial monitoring during build and 
+ * debug stages) for the platform has been initialized.
+ */
+void init_uart(void);
```

Comment:
Any updates on this thought? I can remove it and try testing solely via the debugger (ST Link) for the time being?

Reply:
I think you could keep the changes locally for testing purposes, and then just omit them from the PR. I see the value of the UART here, but it would be nice to contain the scope as much as possible.